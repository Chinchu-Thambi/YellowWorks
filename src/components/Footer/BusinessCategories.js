import React from 'react';
import PropTypes from 'prop-types';
import { Flex } from 'rebass/styled-components';

import tradesServices from './assets/trades-services.svg';
import automotive from './assets/automotive.svg';
import healthBeauty from './assets/health-beauty.svg';
import medical from './assets/medical.svg';
import foodDrink from './assets/food-drink.svg';
import travel from './assets/travel.svg';
import community from './assets/community.svg';
import retailShopping from './assets/retail-shopping.svg';

import { UpperFooterStyle } from './styledComponents';

export const BusinessCategories = ({ children }) => (
  <UpperFooterStyle state="true">
    <div>
      <h2>
        Popular business categories
        {/* <button
          type="button"
          onClick={toggle}
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab={toggleGALab}
        >
          <img src={arrowDown} alt={toggleArrowAlt} />
        </button> */}
      </h2>
    </div>
    { children }
  </UpperFooterStyle>
);

export const BusinessCategoriesContent = ({ theme }) => (
  <Flex
    justifyContent="center"
    mx="auto"
    width={theme.containerWidth}
    flexWrap="wrap"
  >
    <ul>
      <li>
        <a
          href="/auckland/trades-services?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Trades_&_Services"
        >
          <img src={tradesServices} alt="trades-services" />
          <h3>Trades & services</h3>
        </a>
      </li>
      <li>
        <a
          href="/auckland/plumbers?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Trades_&_Services_Plumbers"
        >
          Plumbers
        </a>
      </li>
      <li>
        <a
          href="/auckland/builders?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Trades_&_Services_Builders"
        >
          Builders
        </a>
      </li>
      <li>
        <a
          href="/auckland/electricians?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Trades_&_Services_Electricians"
        >
          Electricians
        </a>
      </li>
    </ul>
    <ul>
      <li>
        <a
          href="/auckland/automotive?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Automotive"
        >
          <img src={automotive} alt="automotive" />
          <h3>Automotive</h3>
        </a>
      </li>
      <li>
        <a
          href="/auckland/automotive?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Automotive_Repairs"
        >
          Repairs
        </a>
      </li>
      <li>
        <a
          href="/auckland/tyres?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Automotive_Tyres"
        >
          Tyres
        </a>
      </li>
      <li>
        <a
          href="/auckland/dismantlers?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Automotive_Dismantlers"
        >
          Dismantlers
        </a>
      </li>
    </ul>
    <ul>
      <li>
        <a
          href="/auckland/health-beauty?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Health_&_Beauty"
        >
          <img src={healthBeauty} alt="health-beauty" />
          <h3>Health & beauty</h3>
        </a>
      </li>
      <li>
        <a
          href="/auckland/hairdressers?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Health_&_Beauty_Hairdressers"
        >
          Hairdressers
        </a>
      </li>
      <li>
        <a
          href="/auckland/beauty-therapy?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Health_&_Beauty_Beauty_Therapy"
        >
          Beauty therapy
        </a>
      </li>
      <li>
        <a
          href="/auckland/health-fitness-centres?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Health_&_Beauty_Health_&_Fitness_Centres"
        >
          Health & fitness centres
        </a>
      </li>
    </ul>
    <ul>
      <li>
        <a
          href="/auckland/medical?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Medical"
        >
          <img src={medical} alt="medical" />
          <h3>Medical</h3>
        </a>
      </li>
      <li>
        <a
          href="/auckland/dentists?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Medical_Dentists"
        >
          Dentists
        </a>
      </li>
      <li>
        <a
          href="/auckland/pharmacies?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Medical_Pharmacies"
        >
          Pharmacies
        </a>
      </li>
      <li>
        <a
          href="/auckland/medical-clinics?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Medical_Medical_Clinics"
        >
          Medical clinics
        </a>
      </li>
    </ul>
    <ul>
      <li>
        <a
          href="/auckland/food-drink-dining?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Food_&_Drink"
        >
          <img src={foodDrink} alt="food-drink" />
          <h3>Food & drink</h3>
        </a>
      </li>
      <li>
        <a
          href="/auckland/restaurants?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Food_&_Drink_Restaurants"
        >
          Restaurants
        </a>
      </li>
      <li>
        <a
          href="/auckland/takeaway-foods?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Food_&_Drink_Takeaway_Foods"
        >
          Takeaway foods
        </a>
      </li>
      <li>
        <a
          href="/auckland/cafes?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Food_&_Drink_Cafes"
        >
          Cafes
        </a>
      </li>
    </ul>
    <ul>
      <li>
        <a
          href="/auckland/travel-accommodation?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Travel"
        >
          <img src={travel} alt="travel" />
          <h3>Travel</h3>
        </a>
      </li>
      <li>
        <a
          href="/auckland/hotels?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Travel_Hotel"
        >
          Hotel
        </a>
      </li>
      <li>
        <a
          href="/auckland/rental?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Travel_Rental"
        >
          Rental
        </a>
      </li>
      <li>
        <a
          href="/auckland/motels-lodges?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Travel_Motels_&_Lodges"
        >
          Motels & lodges
        </a>
      </li>
    </ul>
    <ul>
      <li>
        <a
          href="/auckland/community?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Community"
        >
          <img src={community} alt="community" />
          <h3>Community</h3>
        </a>
      </li>
      <li>
        <a
          href="/auckland/justices-of-the-peace?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Community_Justices_Of_The_Peace"
        >
          Justices of the peace
        </a>
      </li>
      <li>
        <a
          href="/auckland/information-centres?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Community_Information_Centres"
        >
          Information centres
        </a>
      </li>
      <li>
        <a
          href="/auckland/churches?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Community_Churches"
        >
          Churches
        </a>
      </li>
    </ul>
    <ul>
      <li>
        <a
          href="/auckland/retail-shopping?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Retail_&_Shopping"
        >
          <img src={retailShopping} alt="retail-shopping" />
          <h3>Retail & shopping</h3>
        </a>
      </li>
      <li>
        <a
          href="/auckland/florists?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Retail_&_Shopping_Florists"
        >
          Florists
        </a>
      </li>
      <li>
        <a
          href="/auckland/bookshops?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Retail_&_Shopping_Bookshops"
        >
          Bookshops
        </a>
      </li>
      <li>
        <a
          href="/auckland/pet-shops?pop_category=true"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Business_Category_Retail_&_Shopping_Pet_Shops"
        >
          Pet shops
        </a>
      </li>
    </ul>
  </Flex>
);

const themePropType = PropTypes.shape({
  containerWidth: PropTypes.arrayOf(
    PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
  ),
});

BusinessCategoriesContent.propTypes = {
  theme: themePropType.isRequired,
};

BusinessCategories.propTypes = {
  children: PropTypes.node.isRequired,
};
