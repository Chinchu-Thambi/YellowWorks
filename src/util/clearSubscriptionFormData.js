/* globals localStorage */

import { storageAvailable } from '.';

const clearSubscriptionFormData = () => {
  if (storageAvailable()) {
    Object.keys(localStorage).forEach((key) => {
      if (
        key === 'storageOrderDetails'
        || key.startsWith('storageFormData')
        || key.startsWith('formData-') // still needed because onboarding uses it, yet.
      ) {
        localStorage.removeItem(key);
      }
    });
  }
};

export default clearSubscriptionFormData;
