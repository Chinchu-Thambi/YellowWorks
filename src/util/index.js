/* global window, DOMException, document */

import React from 'react';
import showdown from 'showdown';

import { importableModules } from '../services/useModule';

export {
  default as theme,
  mediaBreakpointUp,
} from './theme';
export {
  default as errorText,
  infos as infoText,
  getErrorString,
} from './texts';

const isSingleLine = (string) => (string !== undefined ? !string.includes('\n') : '');

const mdConverter = new showdown.Converter();

mdConverter.setOption('tables', true);

export const parseMd = (md) => {
  if (!md) {
    return '';
  }

  let mdString = md;

  if (typeof md === 'object') {
    [mdString] = Object.values(md);
  }

  let parsedString = mdConverter.makeHtml(mdString);

  if (isSingleLine(mdString)) {
    // remove outer p tag from single line md
    parsedString = parsedString.replace(/<\/?p>/g, '');
  }

  // eslint-disable-next-line react/no-danger
  return <span dangerouslySetInnerHTML={{ __html: parsedString }} />;
};

export const getProportionalHeight = ({
  initialWidth, initialHeight, targetWidth,
}) => (initialHeight * targetWidth) / initialWidth;

export const storageAvailable = () => {
  let storage;

  if (typeof window !== 'undefined') {
    storage = window.localStorage;
  }

  try {
    const x = '__storage_test__';
    storage.setItem(x, x);
    storage.removeItem(x);
    return true;
  } catch (e) {
    if (typeof DOMException === 'undefined') {
      return false;
    }

    return e instanceof DOMException && (
      // everything except Firefox
      e.code === 22
      // Firefox
      || e.code === 1014
      // test name field too, because code might not be present
      // everything except Firefox
      || e.name === 'QuotaExceededError'
      // Firefox
      || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
      // acknowledge QuotaExceededError only if there's something already stored
      && storage && storage.length !== 0;
  }
};

export const formatPrice = (price = 0) => (typeof price === 'number' ? `$${price.toFixed(2)}` : `$${price}`);

export const formatRecurringPrice = (price, recurring, withGst, annual) => {
  const priceFormatted = formatPrice(price);
  if (priceFormatted === '$0.00') {
    return 'Free';
  }

  if (recurring) {
    if (withGst) {
      return `${priceFormatted} / month + GST`;
    }
    if (annual) {
      return `${priceFormatted} / year`;
    }
    return `${priceFormatted} / month`;
  }
  if (withGst) {
    return `${priceFormatted} + GST`;
  }
  return priceFormatted;
};

// ex input date object
// ex out: "13/08/2019"
export const reformatDate = (dateObj) => (
  dateObj ? dateObj.toLocaleDateString('en-NZ') : ''
);

export const useLocalStorage = (key, initialValue) => {
  // from https://usehooks.com/useLocalStorage/
  // State to store our value
  // Pass initial state function to useState so logic is only executed once
  const [storedValue, setStoredValue] = React.useState(() => {
    try {
      // Get from local storage by key
      if (storageAvailable()) {
        const item = window.localStorage.getItem(key);
        // Parse stored json or if none return initialValue
        return item ? JSON.parse(item) : initialValue;
      }
      return initialValue;
    } catch (error) {
      // If error also return initialValue
      console.log(error);
      return initialValue;
    }
  });

  // Return a wrapped version of useState's setter function that ...
  // ... persists the new value to localStorage.
  const setValue = (value) => {
    try {
      // Allow value to be a function so we have same API as useState
      const valueToStore = value instanceof Function ? value(storedValue) : value;
      // Save state
      setStoredValue(valueToStore);
      // Save to local storage
      if (storageAvailable()) {
        window.localStorage.setItem(key, JSON.stringify(valueToStore));
      }
    } catch (error) {
      // A more advanced implementation would handle the error case
      console.log(error);
    }
  };

  return [storedValue, setValue];
};

// TODO: remove this and user version inside of auth/services
export const isStaff = async () => {
  try {
    const { Auth } = await importableModules['@aws-amplify/auth']();
    const user = await Auth.currentSession();
    return user.idToken.payload.email.endsWith('@yellow.co.nz');
  } catch (err) {
    // console.log('isStaff', err);
    return false;
  }
};


export const isTesting = () => document.location.search.includes('utm_isTesting=yes');

export const getNextWeekday = () => {
  const dt = new Date();
  dt.setDate(dt.getDate() + 1);

  if (dt.getDay() === 6) { // if Saturday
    dt.setDate(dt.getDate() + 2);
  }

  if (dt.getDay() === 0) { // if Sunday
    dt.setDate(dt.getDate() + 1);
  }

  return dt;
};

export const getNextMonth = ({
  baseNextWeekday = false,
} = {}) => {
  let dt = new Date();

  if (baseNextWeekday) {
    dt = getNextWeekday();
  }

  dt.setFullYear(
    dt.getFullYear(),
    dt.getMonth() + 1,
    1,
  );

  return dt;
};

export const getEndOfMonth = ({
  baseNextWeekday = false,
} = {}) => {
  let dt = new Date();

  if (baseNextWeekday) {
    dt = getNextWeekday();
  }

  dt.setFullYear(
    dt.getFullYear(),
    dt.getMonth() + 1,
    0,
  );

  return dt;
};

export const capitalize = (string = '') => `${string.charAt(0).toUpperCase()}${string.substr(1).toLowerCase()}`;

export const shuffle = (array) => {
  let counter = array.length;
  const shuffled = [...array];

  // While there are elements in the array
  while (counter > 0) {
    // Pick a random index
    const index = Math.floor(Math.random() * counter);

    // Decrease counter by 1
    counter -= 1;

    // And swap the last element with it
    const temp = shuffled[counter];
    shuffled[counter] = shuffled[index];
    shuffled[index] = temp;
  }

  return shuffled;
};
