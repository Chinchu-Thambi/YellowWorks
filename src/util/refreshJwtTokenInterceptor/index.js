import refreshJwtToken from './refreshJwtTokenInterceptor';

export default refreshJwtToken;
