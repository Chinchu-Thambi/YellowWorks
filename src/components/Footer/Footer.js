import React from 'react';
import PropTypes from 'prop-types';
import { Flex, Box } from 'rebass/styled-components';
import styled, { withTheme } from 'styled-components';

import { mediaBreakpointUp, theme } from '../../util';

import { StyledFooter, MainFooter } from './styledComponents';
import { BusinessCategories, BusinessCategoriesContent } from './BusinessCategories';

const MainFooterStyled = styled(Flex)`
  h3 {
    font-size: ${theme.fontSizes[2]};
    font-weight: ${theme.fontWeight[1]};
  }
  > ul {
    flex-grow: 1;
    
    a {
      font-size: ${theme.fontSizes[1]};
      display: block;
      padding-top: .75em;
    }
  }
`;

const HiddenOnMobileUl = styled(Box)`
  display: none;

  ${mediaBreakpointUp('md')} {
    display: block;
  }
`;

const MainFooterContent = () => (
  <MainFooterStyled
    justifyContent="space-between"
    mx="auto"
    pb={[null, null, 3]}
  >
    <HiddenOnMobileUl as="ul">
      <li><h3>Services</h3></li>
      <li>
        <a
          href="/our-products/yellow-profile/"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Services_Yellow_Profile"
        >
          Yellow Profile
        </a>
      </li>
      <li>
        <a
          href="/our-products/seo/"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Services_Search_Engine_Optimisation"
        >
          Search Engine Optimisation
        </a>
      </li>
      <li>
        <a
          href="/our-products/google-search-ads/"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Services_Google_AdWords"
        >
          Google Search Ads
        </a>
      </li>
      <li>
        <a
          href="/our-products/google-display-ads/"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Services_Display_Advertising"
        >
          Google Display Ads
        </a>
      </li>
      <li>
        <a
          href="/our-products/facebook-ads/"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Services_Facebook_Content"
        >
          Facebook Ads
        </a>
      </li>
      <li>
        <a
          href="/our-products/marketing-automation/"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Services_Marketing_Automation"
        >
          Marketing Automation & CRM
        </a>
      </li>
      <li>
        <a
          href="/our-products/yellow-pages/"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Services_Yellow_Pages"
        >
          Yellow Pages
        </a>
      </li>
      <li>
        <a
          href="/our-products/white-pages/"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Services_White_Pages"
        >
          White Pages
        </a>
      </li>
    </HiddenOnMobileUl>
    <Box as="ul">
      <li><h3>Sites</h3></li>
      <li>
        <a
          href="https://whitepages.co.nz/"
          target="_blank"
          rel="noopener noreferrer"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Our_sites_White"
        >
          White
        </a>
      </li>
      <li>
        <a
          href="https://www.finda.co.nz/"
          target="_blank"
          rel="noopener noreferrer"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Our_sites_Finda"
        >
          Finda
        </a>
      </li>
      <li>
        <a
          href="https://www.menus.co.nz/"
          target="_blank"
          rel="noopener noreferrer"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Our_sites_Menus"
        >
          Menus
        </a>
      </li>
      <li>
        <a
          href="https://www.tourism.net.nz/"
          target="_blank"
          rel="noopener noreferrer"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Our_sites_NZ_Tourism_Guide"
        >
          NZ Tourism Guide
        </a>
      </li>
      <li>
        <a
          href="https://yellow.co.nz/018/"
          target="_blank"
          rel="noopener noreferrer"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Our_sites_Call_018"
        >
          Call 018
        </a>
      </li>
    </Box>
    <HiddenOnMobileUl as="ul">
      <li><h3>Resources</h3></li>
      <li>
        <a
          href="/resources/customer-stories/"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Clients_Customer_Stories"
        >
          Customer Stories
        </a>
      </li>
      <li>
        <a
          href="/resources/tips-and-tools/"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Tips and Tools"
        >
          Tips and Tools
        </a>
      </li>
      <li>
        <a
          href="/resources/yellow-ebook/"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Ebook"
        >
          Yellow eBook
        </a>
      </li>
      <li>
        <a
          href="https://hub.yellow.co.nz/knowledge/faq"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="FAQ"
        >
          FAQ
        </a>
      </li>
      <li>
        <a
          href="/resources/flybuys/"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Header_Clicks"
          data-ga-lab="Resources_Flybuys"
        >Flybuys
        </a>
      </li>
    </HiddenOnMobileUl>
    <HiddenOnMobileUl as="ul">
      <li><h3>Company</h3></li>
      <li>
        <a
          href="/company/about-us/"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Our_Team"
        >
          Our Team
        </a>
      </li>
      <li>
        <a
          href="/company/careers/"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Careers"
        >
          Careers
        </a>
      </li>
      <li>
        <a
          href="/company/contact-us/"
          data-ga="track"
          data-ga-cat="Navigation_Interaction"
          data-ga-act="Footer_Clicks"
          data-ga-lab="Contact_Us"
        >
          Contact Us
        </a>
      </li>
    </HiddenOnMobileUl>
  </MainFooterStyled>
);

MainFooterContent.propTypes = {
};

const Footer = ({ hideCategories }) => (
  <StyledFooter>
    {!hideCategories && (
      <BusinessCategories>
        <BusinessCategoriesContent theme={theme} />
      </BusinessCategories>
    )}
    <MainFooter theme={theme}>
      <MainFooterContent theme={theme} />
    </MainFooter>
  </StyledFooter>
);

Footer.defaultProps = {
  hideCategories: false,
};

Footer.propTypes = {
  hideCategories: PropTypes.bool,
};

export default withTheme(Footer);
