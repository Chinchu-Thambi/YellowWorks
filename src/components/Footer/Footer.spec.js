/* globals describe, it, expect, jest */

import React from 'react';
import { render } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import $ from 'cheerio';

import { theme } from '../../util';

import Footer from '.';

jest.mock('react-shielded', () => ({ __esModule: true, default: () => null }));

describe('Footer', () => {
  it('renders the correct list of links', () => {
    const wrapper = render(
      <ThemeProvider theme={theme}>
        <Footer />
      </ThemeProvider>,
    );

    const links = wrapper
      .find('a')
      .map((index, link) => ({
        href: $(link).attr('href'),
        html: $(link).text(),
      }))
      .toArray();

    expect(links).toMatchSnapshot();
  });

  it('hides business categories based on a prop', () => {
    const wrapper = render(
      <ThemeProvider theme={theme}>
        <Footer hideCategories />
      </ThemeProvider>,
    );

    const categoriesHeading = wrapper.find('h2');

    expect(categoriesHeading).toHaveLength(0);
  });
});
