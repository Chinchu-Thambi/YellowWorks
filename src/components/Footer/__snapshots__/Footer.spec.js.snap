// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders the correct list of links 1`] = `
Array [
  Object {
    "href": "/auckland/trades-services?pop_category=true",
    "html": "Trades & services",
  },
  Object {
    "href": "/auckland/plumbers?pop_category=true",
    "html": "Plumbers",
  },
  Object {
    "href": "/auckland/builders?pop_category=true",
    "html": "Builders",
  },
  Object {
    "href": "/auckland/electricians?pop_category=true",
    "html": "Electricians",
  },
  Object {
    "href": "/auckland/automotive?pop_category=true",
    "html": "Automotive",
  },
  Object {
    "href": "/auckland/automotive?pop_category=true",
    "html": "Repairs",
  },
  Object {
    "href": "/auckland/tyres?pop_category=true",
    "html": "Tyres",
  },
  Object {
    "href": "/auckland/dismantlers?pop_category=true",
    "html": "Dismantlers",
  },
  Object {
    "href": "/auckland/health-beauty?pop_category=true",
    "html": "Health & beauty",
  },
  Object {
    "href": "/auckland/hairdressers?pop_category=true",
    "html": "Hairdressers",
  },
  Object {
    "href": "/auckland/beauty-therapy?pop_category=true",
    "html": "Beauty therapy",
  },
  Object {
    "href": "/auckland/health-fitness-centres?pop_category=true",
    "html": "Health & fitness centres",
  },
  Object {
    "href": "/auckland/medical?pop_category=true",
    "html": "Medical",
  },
  Object {
    "href": "/auckland/dentists?pop_category=true",
    "html": "Dentists",
  },
  Object {
    "href": "/auckland/pharmacies?pop_category=true",
    "html": "Pharmacies",
  },
  Object {
    "href": "/auckland/medical-clinics?pop_category=true",
    "html": "Medical clinics",
  },
  Object {
    "href": "/auckland/food-drink-dining?pop_category=true",
    "html": "Food & drink",
  },
  Object {
    "href": "/auckland/restaurants?pop_category=true",
    "html": "Restaurants",
  },
  Object {
    "href": "/auckland/takeaway-foods?pop_category=true",
    "html": "Takeaway foods",
  },
  Object {
    "href": "/auckland/cafes?pop_category=true",
    "html": "Cafes",
  },
  Object {
    "href": "/auckland/travel-accommodation?pop_category=true",
    "html": "Travel",
  },
  Object {
    "href": "/auckland/hotels?pop_category=true",
    "html": "Hotel",
  },
  Object {
    "href": "/auckland/rental?pop_category=true",
    "html": "Rental",
  },
  Object {
    "href": "/auckland/motels-lodges?pop_category=true",
    "html": "Motels & lodges",
  },
  Object {
    "href": "/auckland/community?pop_category=true",
    "html": "Community",
  },
  Object {
    "href": "/auckland/justices-of-the-peace?pop_category=true",
    "html": "Justices of the peace",
  },
  Object {
    "href": "/auckland/information-centres?pop_category=true",
    "html": "Information centres",
  },
  Object {
    "href": "/auckland/churches?pop_category=true",
    "html": "Churches",
  },
  Object {
    "href": "/auckland/retail-shopping?pop_category=true",
    "html": "Retail & shopping",
  },
  Object {
    "href": "/auckland/florists?pop_category=true",
    "html": "Florists",
  },
  Object {
    "href": "/auckland/bookshops?pop_category=true",
    "html": "Bookshops",
  },
  Object {
    "href": "/auckland/pet-shops?pop_category=true",
    "html": "Pet shops",
  },
  Object {
    "href": "/our-products/yellow-profile/",
    "html": "Yellow Profile",
  },
  Object {
    "href": "/our-products/seo/",
    "html": "Search Engine Optimisation",
  },
  Object {
    "href": "/our-products/google-search-ads/",
    "html": "Google Search Ads",
  },
  Object {
    "href": "/our-products/google-display-ads/",
    "html": "Google Display Ads",
  },
  Object {
    "href": "/our-products/facebook-ads/",
    "html": "Facebook Ads",
  },
  Object {
    "href": "/our-products/marketing-automation/",
    "html": "Marketing Automation & CRM",
  },
  Object {
    "href": "/our-products/yellow-pages/",
    "html": "Yellow Pages",
  },
  Object {
    "href": "/our-products/white-pages/",
    "html": "White Pages",
  },
  Object {
    "href": "https://whitepages.co.nz/",
    "html": "White",
  },
  Object {
    "href": "https://www.finda.co.nz/",
    "html": "Finda",
  },
  Object {
    "href": "https://www.menus.co.nz/",
    "html": "Menus",
  },
  Object {
    "href": "https://www.tourism.net.nz/",
    "html": "NZ Tourism Guide",
  },
  Object {
    "href": "https://yellow.co.nz/018/",
    "html": "Call 018",
  },
  Object {
    "href": "/resources/customer-stories/",
    "html": "Customer Stories",
  },
  Object {
    "href": "/resources/tips-and-tools/",
    "html": "Tips and Tools",
  },
  Object {
    "href": "/resources/yellow-ebook/",
    "html": "Yellow eBook",
  },
  Object {
    "href": "https://hub.yellow.co.nz/knowledge/faq",
    "html": "FAQ",
  },
  Object {
    "href": "/resources/flybuys/",
    "html": "Flybuys",
  },
  Object {
    "href": "/company/about-us/",
    "html": "Our Team",
  },
  Object {
    "href": "/company/careers/",
    "html": "Careers",
  },
  Object {
    "href": "/company/contact-us/",
    "html": "Contact Us",
  },
  Object {
    "href": "tel:0800803803",
    "html": "0800 803 803",
  },
  Object {
    "href": "mailto:care@yellow.co.nz",
    "html": "care@yellow.co.nz",
  },
  Object {
    "href": "https://www.facebook.com/yellownz",
    "html": "",
  },
  Object {
    "href": "https://twitter.com/yellownz",
    "html": "",
  },
  Object {
    "href": "https://www.linkedin.com/company/yellow-new-zealand",
    "html": "",
  },
  Object {
    "href": "https://www.youtube.com/user/yellowbiznz",
    "html": "",
  },
  Object {
    "href": "https://www.google.co.nz/maps/place/Yellow/@-36.897607,174.8006043,17z/data=!3m1!4b1!4m5!3m4!1s0x6d0d48ef0f099ad1:0xbabdfef7f73a92a5!8m2!3d-36.897607!4d174.802793",
    "html": "",
  },
  Object {
    "href": "/terms/privacy-policy/",
    "html": "Privacy policy",
  },
  Object {
    "href": "/terms/",
    "html": "Legal",
  },
]
`;
